pkgs/os-specific/linux/service-wrapper/default.nix

```diff
@@ -0,0 +1,280 @@
+{ stdenv, fetchurl, systemd, coreutils,
+  src ? null }:
+
+stdenv.mkDerivation rec {
+  name = "service-wrapper-${version}";
+  version = "16.04.0"; # Ajar to Ubuntu Release
+
+  inherit src;
```

Comment:
Better would be to separate real `service-wrapper` to separate file and use it in src:

`src = ./service-wrapper.sh;`

You don't need to pass src as attribute because you can directly override derivation attributes (`lib.overrideDerivation`) later as you wish.